### PR TITLE
Google firebase auth

### LINKED ISSUE #26: Create log-in screen

### PR DESCRIPTION
**Pull request by evguenirousselot**
For this pull request several things were put in place :

	1) UI for the login page with logo and signing in with google
	2) Link to Firebase to enable google authentication
	3) Signing out with google
	4) Caching of the user's credentials
	5) Refactoring with Jetpack Compose

Notes :

	-> Signing In was done with the official documentation and deprecated methods were replaced with their new equivalent

	-> The Sign Out button is still in xml because it needs to later be incorporated to the Jetpack compose dashboard/profile design

	-> Automated testing limits itself to checking that the two buttons are present and can be clicked on, because of Google's API and the absence of caching on the emulators, correct and incorrect input of credentials was manually tested

This closes #26 , #27 and #28

### CODE REVIEW

File: app/google-services.json [hunk 1]
```diff
@@ -6,35 +6,6 @@
     "storage_bucket": "sdp-orkest-firebase.appspot.com"
   },
   "client": [
-    {
```
Comment (yasmineben510) [inline app/google-services.json:9]: Why was this deletion needed?

File: app/src/androidTest/java/com/github/orkest/SignOutTest.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package com.github.orkest
```
Comment (yasmineben510) [inline app/src/androidTest/java/com/github/orkest/SignOutTest.kt:1]: These tests look good to me! Nice job

File: app/src/androidTest/java/com/github/orkest/SignInTest.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package com.github.orkest
```
Comment (yasmineben510) [inline app/src/androidTest/java/com/github/orkest/SignInTest.kt:1]: Nice tests, an idea would be to add a test to check if the logo image is indeed displayed as expected on the UI

File: app/src/main/java/com/github/orkest/MainActivity.kt [hunk 4]
```diff
@@ -1,17 +1,26 @@
 package com.github.orkest
```
Comment (yasmineben510) [inline app/src/main/java/com/github/orkest/MainActivity.kt:1]: Is the signUp incorporated with this signIn or is the signUp with google is yet to be implemented

File: app/src/main/java/com/github/orkest/ui/signin/SignIn.kt [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                LogoImage()
+                SignInButton()
+
+            }
+        }
+
+        // Configure Google Sign In
```
Comment (yasmineben510) [inline app/src/main/java/com/github/orkest/ui/signin/SignIn.kt:62]: Maybe this part could be refactored into a viewModel class associated to this UI

**Review by yasmineben510 — COMMENTED** [review]
This is a nice work thank you ! 
A few remarks would be:
- Try to implement it using the MVVM pattern by separating the UI from the state 
- Try to merge the new main in your branch to resolve the current conflicts that are currently resolvable to allow you to implement the complete compose refactoring since the main is now up to date with respect to that along with linking it to the current authentication logic being implemented

File: app/src/main/java/com/github/orkest/ui/signin/SignIn.kt [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+import com.google.firebase.auth.GoogleAuthProvider
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
+
+
+/**
+ * A login screen that offers login via Google with Firebase
+ * @see [GoogleSignInFirebase](https://firebase.google.com/docs/auth/android/firebaseui?hl=fr)
+ */
+class SignIn : ComponentActivity() {
```
Comment (emilehreich) [inline app/src/main/java/com/github/orkest/ui/signin/SignIn.kt:40]: When the Sign In is successful, the app doesn't seem to launch the MainActivity. A successful sign in should probably direct the user to the profile or feed page.

**Review by emilehreich — COMMENTED** [review]
How will this be linked to the signup page (@yasmineben510 's work)? Will they be merged or completely independent?

File: app/google-services.json (hunk 1 above)
Comment (evguenirousselot) [reply app/google-services.json:9]: the previous ones were outdated

File: app/src/androidTest/java/com/github/orkest/SignInTest.kt (hunk 3 above)
Comment (evguenirousselot) [reply app/src/androidTest/java/com/github/orkest/SignInTest.kt:1]: thanks I'll add this

File: app/src/main/java/com/github/orkest/ui/signin/SignIn.kt (hunk 5 above)
Comment (evguenirousselot) [reply app/src/main/java/com/github/orkest/ui/signin/SignIn.kt]: Yes I've refactored my entire code with the MVVM model

File: app/src/main/java/com/github/orkest/ui/signin/SignIn.kt (hunk 6 above)
Comment (evguenirousselot) [reply app/src/main/java/com/github/orkest/ui/signin/SignIn.kt]: Now it redirects to the the signup. Main was just for testing

File: app/src/main/java/com/github/orkest/MainActivity.kt (hunk 4 above)
Comment (evguenirousselot) [reply app/src/main/java/com/github/orkest/MainActivity.kt]: the signup will be in between the two

**Review by yasmineben510 — APPROVED** [review]

**Pull request merged**